app/services/atmosphere/cloud/satisfy_appliance.rb

```diff
@@ -0,0 +1,88 @@
+module Atmosphere
+  module Cloud
+    class SatisfyAppliance
+      def initialize(appliance)
+        @appliance = appliance
+      end
+
+      def execute
+        action.log(I18n.t('satisfy_appliance.start'))
+        if appliance.virtual_machines.blank?
+          if optimization_strategy.can_reuse_vm? &&
+            !(vm = optimization_strategy.vm_to_reuse).nil?
+
+            action.log(I18n.t('satisfy_appliance.reusing', vm: vm.id_at_site))
+            appl_manager.reuse_vm!(vm)
+            action.log(I18n.t('satisfy_appliance.reused', vm: vm.id_at_site))
+          else
+            new_vms_tmpls_and_flavors.each do |tmpl_and_flavor|
+              tmpl = tmpl_and_flavor[:template]
+              flavor = tmpl_and_flavor[:flavor]
+              if tmpl.blank?
+                appliance.state = :unsatisfied
+                appliance.state_explanation =
+                  I18n.t('satisfy_appliance.no_tmpl')
+
+                action.warn(appliance.state_explanation)
+              elsif flavor.nil?
+                appliance.state = :unsatisfied
+                appliance.state_explanation =
+                  I18n.t('satisfy_appliance.no_flavor')
+
+                action.warn(appliance.state_explanation)
+              else
+                action.log(I18n.t('satisfy_appliance.starting_vm',
+                                  tmpl: tmpl.id_at_site,
+                                  flavor: flavor.id_at_site))
+                appl_manager.spawn_vm!(tmpl, flavor, vm_name)
+                action.log(I18n.t('satisfy_appliance.vm_started',
+                                  tmpl: tmpl.id_at_site,
+                                  flavor: flavor.id_at_site))
+              end
+            end
+          end
+          unless appl_manager.save
```

Comment:
That is a good question :smile: I believe that user will have free VM until next billing execution is triggered. After that (s)he will be charged for previous hour. 

Of course this is a bug (https://github.com/dice-cyfronet/atmosphere/issues/174) and we should fix it. I don't want to fix is in the scope of this PR, because here I'm just moving code around files.